quickstarts/microsoft.network/application-gateway-v2-autoscale-create/main.bicep

```diff
@@ -0,0 +1,172 @@
+@description('Virtual Network name')
+param virtualNetworkName string = 'Application-Vnet'
+
+@description('Virtual Network address range')
+param vnetAddressPrefix string = '10.0.0.0/16'
+
+@description('Name of the subnet')
+param subnetName string = 'ApplicationGatewaySubnet'
+
+@description('Subnet address range')
+param subnetPrefix string = '10.0.0.0/24'
+
+@description('Application Gateway name')
+param applicationGatewayName string = 'applicationGatewayV2'
+
+@description('Minimum instance count for Application Gateway')
+param minCapacity int = 2
+
+@description('Maximum instance count for Application Gateway')
+param maxCapacity int = 10
+
+@description('Application Gateway Frontend port')
+param frontendPort int = 80
+
+@description('Application gateway Backend port')
+param backendPort int = 80
+
+@description('Back end pool ip addresses')
+param backendIPAddresses array = [
+  {
+    IpAddress: '10.0.0.4'
+  }
+  {
+    IpAddress: '10.0.0.5'
+  }
+]
+
+@allowed([
+  'Enabled'
+  'Disabled'
+])
+@description('Cookie based affinity')
+param cookieBasedAffinity string = 'Disabled'
+
+@description('Location for all resources.')
+param location string = resourceGroup().location
+
+var appGwPublicIpName = '${applicationGatewayName}-pip'
+var appGwSize = 'Standard_v2'
+
+resource virtualNetwork 'Microsoft.Network/virtualNetworks@2020-06-01' = {
+  name: virtualNetworkName
+  location: location
+  properties: {
+    addressSpace: {
+      addressPrefixes: [
+        vnetAddressPrefix
+      ]
+    }
+    subnets: [
+      {
+        name: subnetName
+        properties: {
+          addressPrefix: subnetPrefix
+        }
+      }
+    ]
+  }
+}
+
+resource publicIP 'Microsoft.Network/publicIPAddresses@2020-06-01' = {
+  name: appGwPublicIpName
+  location: location
+  sku: {
+    name: 'Standard'
+  }
+  properties: {
+    publicIPAllocationMethod: 'Static'
+  }
+}
+
+resource applicationGateway 'Microsoft.Network/applicationGateways@2020-06-01' = {
+  name: applicationGatewayName
+  location: location
+  properties: {
+    sku: {
+      name: appGwSize
+      tier: 'Standard_v2'
+    }
+    autoscaleConfiguration: {
+      minCapacity: minCapacity
+      maxCapacity: maxCapacity
+    }
+    gatewayIPConfigurations: [
+      {
+        name: 'appGatewayIpConfig'
+        properties: {
+          subnet: {
+            id: resourceId('Microsoft.Network/virtualNetworks/subnets', virtualNetwork.name, subnetName)
```

Comment:
bicep-ify?

Reply:
Not sure how to do that here.

Reply:
use subnet.id - I think bicep might still required the subnet to be defined as a child resource first...